src/pages/iou/steps/IOUAmountPage.js

```diff
@@ -28,25 +50,83 @@ const propTypes = {
 const defaultProps = {
     iou: {},
 };
+class IOUAmountPage extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            textInputWidth: 0,
+        };
+    }
 
-const IOUAmountPage = props => (
-    <View style={styles.pageWrapper}>
-        {props.iou.loading && <ActivityIndicator color={themeColors.text} />}
-        <TouchableOpacity
-            style={[styles.button, styles.w100, styles.mt5]}
-            onPress={props.onStepComplete}
-        >
-            <Text style={[styles.buttonText]}>
-                Next
-            </Text>
-        </TouchableOpacity>
-    </View>
-);
+    componentDidMount() {
+        if (this.textInput) {
+            this.textInput.focus();
+        }
+    }
 
+    render() {
+        return (
+            <View style={[styles.flex1, styles.pageWrapper]}>
+                {this.props.iou.loading && <ActivityIndicator color={themeColors.text} />}
+                <View style={[
+                    styles.flex1,
+                    styles.flexRow,
+                    styles.w100,
+                    styles.alignItemsCenter,
+                    styles.justifyContentCenter,
+                ]}
+                >
+                    <Text style={styles.iouAmountText}>
+                        {this.props.selectedCurrency}
+                    </Text>
+                    {this.props.isSmallScreenWidth
+                        ? <Text style={styles.iouAmountText}>{this.props.amount}</Text>
+                        : (
+                            <View>
+                                <TextInputFocusable
+                                        style={[styles.iouAmountTextInput,
+                                            {width: Math.max(5, this.state.textInputWidth)}]}
+                                        onKeyPress={(event) => {
+                                            this.props.numberPressed(event.key);
+                                            event.preventDefault();
+                                        }}
+                                        ref={el => this.textInput = el}
+                                        defaultValue={this.props.amount}
+                                        textAlign="left"
+                                />
+                                <Text
+                                    style={[styles.iouAmountText, styles.invisible, {left: 100000}]}
```

Comment:
Please don't use inline styles like `{left: 100000}` directly in the `style` tag. 
Also, why `100000`? It looks maybe like a magic number?